lib/explorer/series.ex

```diff
@@ -2797,6 +2797,155 @@ defmodule Explorer.Series do
+  def upcase(%Series{dtype: :string} = series),
+    do: Shared.apply_impl(series, :upcase)
+
+  def upcase(%Series{dtype: dtype}), do: dtype_error("upcase/1", dtype, [:string])
+
+  @doc """
+  Converts all characters to lowercase.
+
+  ## Examples
+
+      iex> s = Explorer.Series.from_list(["ABC", "DEF", "BCD"])
+      iex> Explorer.Series.downcase(s)
+      #Explorer.Series<
+        Polars[3]
+        string ["abc", "def", "bcd"]
+      >
+  """
+  @doc type: :element_wise
+  @spec downcase(Series.t()) :: Series.t()
+  def downcase(%Series{dtype: :string} = series),
+    do: Shared.apply_impl(series, :downcase)
+
+  def downcase(%Series{dtype: dtype}), do: dtype_error("downcase/1", dtype, [:string])
+
+  @doc """
+  Returns a string where all leading and trailing Unicode whitespaces have been removed.
+
+  ## Examples
+
+      iex> s = Explorer.Series.from_list(["  abc", "def  ", "  bcd"])
+      iex> Explorer.Series.trim(s)
+      #Explorer.Series<
+        Polars[3]
+        string ["abc", "def", "bcd"]
+      >
+  """
+  @doc type: :element_wise
+  @spec trim(Series.t()) :: Series.t()
+  def trim(%Series{dtype: :string} = series),
+    do: Shared.apply_impl(series, :trim)
+
+  def trim(%Series{dtype: dtype}), do: dtype_error("trim/1", dtype, [:string])
+
+  @doc """
+  Returns a string where all leading Unicode whitespaces have been removed.
+
+  ## Examples
+
+      iex> s = Explorer.Series.from_list(["  abc", "def  ", "  bcd"])
+      iex> Explorer.Series.trim_leading(s)
+      #Explorer.Series<
+        Polars[3]
+        string ["abc", "def  ", "bcd"]
+      >
+  """
+  @doc type: :element_wise
+  @spec trim_leading(Series.t()) :: Series.t()
+  def trim_leading(%Series{dtype: :string} = series),
+    do: Shared.apply_impl(series, :trim_leading)
+
+  def trim_leading(%Series{dtype: dtype}), do: dtype_error("trim_leading/1", dtype, [:string])
+
+  @doc """
+  Returns a string where all trailing Unicode whitespaces have been removed.
+
+  ## Examples
+
+      iex> s = Explorer.Series.from_list(["  abc", "def  ", "  bcd"])
+      iex> Explorer.Series.trim_trailing(s)
+      #Explorer.Series<
+        Polars[3]
+        string ["  abc", "def", "  bcd"]
+      >
+  """
+  @doc type: :element_wise
+  @spec trim_trailing(Series.t()) :: Series.t()
+  def trim_trailing(%Series{dtype: :string} = series),
+    do: Shared.apply_impl(series, :trim_trailing)
+
+  def trim_trailing(%Series{dtype: dtype}), do: dtype_error("trim_trailing/1", dtype, [:string])
+
+  @doc """
+  Extracts the regular expression from the string.
+
+  Optionally takes a capture group index.
+
+  ## Examples
+
+      iex> s = Explorer.Series.from_list(["Explorer the library", "is a really", "great library"])
+      iex> Explorer.Series.extract(s, ~r/^(\\w+)\\b/)
+      #Explorer.Series<
+        Polars[3]
+        string ["Explorer", "is", "great"]
+      >
+  """
+  @doc type: :element_wise
+  @spec extract(Series.t(), Regex.t(), pos_integer()) :: Series.t()
```

Comment:
This is called `Regex.scan` in Elixir. However if we call it `scan` it may get confused with `Enum.scan`. Thoughts?

Reply:
Maybe this is a case where it would make sense to prefix the function name? `string_scan`?

Reply:
After more thought, I think scan is fine at the end of the day because it is ultimately different signatures? :) and I think the operations you do with scan are all cumulative operations.

Reply:
So should we keep this function in and just use a `String.t()` for the pattern or should we drop it?

Reply:
I am not sure if it makes sense to scan a static string? It will either return the string itself or nil, so it is basically a contain. So perhaps hold it off until we support regexes which can be next?

Reply:
Yep 👍